alpha/internal-web/app/services/db-client.js

```diff
@@ -1,67 +1,89 @@
 var mongo = require('./database')
+var Promise = require('bluebird')
 
-exports.getAll = function (collectionName, callback) {
-  mongo.db.collection(collectionName).find().toArray(function (error, data) {
-    if (!error) {
-      callback(null, data)
-    } else {
-      callback(error, null)
-    }
+exports.getAll = function (collection) {
+  return new Promise(function (resolve, reject) {
+    mongo.db.collection(collection).find().toArray()
+      .then(function (data) {
+        resolve(data)
+      })
+      .catch(function (error) {
+        reject(error)
+      })
   })
 }
 
-exports.get = function (id, collectionName, callback) {
-  mongo.db.collection(collectionName).findOne({ _id: exports.mongoId(id) }, function (error, data) {
-    if (!error) {
-      callback(null, data)
-    } else {
-      callback(error, null)
-    }
+exports.get = function (id, collection) {
+  return new Promise(function (resolve, reject) {
+    mongo.db.collection(collection).findOne({ _id: exports.mongoId(id) })
+      .then(function (claimant) {
+        resolve(claimant)
+      })
+      .catch(function (error) {
+        reject(error)
+      })
   })
 }
 
-exports.save = function (record, collectionName, callback) {
-  mongo.db.collection(collectionName).insertOne(record, function (error, savedRecord) {
-    if (!error) {
-      callback(null, savedRecord)
-    } else {
-      callback(error, null)
-    }
+exports.save = function (claimant, collection) {
+  return new Promise(function (resolve, reject) {
+    mongo.db.collection(collection).insertOne(claimant)
+      .then(function (savedClaimant) {
+        resolve(savedClaimant.ops[0])
```

Comment:
Why do we now need the property and indexer? The mean of ops[0] isn't really obvious here, though I assume it's introduced by Bluebird?